File: src/main/java/net/kuryshev/webnotes/Note.java
package net.kuryshev.webnotes;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
public class Note {
    @Id
    @SequenceGenerator(name = "note_seq", sequenceName = "note_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "note_seq")
    private Long id;
    private String text;
    private Integer priority;
}

File: src/main/java/net/kuryshev/webnotes/note/Note.kt
package net.kuryshev.webnotes.note

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
data class Note(
        @Id
        @SequenceGenerator(name = "note_seq", sequenceName = "note_seq", allocationSize = 1)
        @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "note_seq")
        var id: Long? = null,
        var title: String = "",
        var text: String = "",
        @Column(name = "project_id")
        var projectId: Long? = null,
        @JsonIgnore
        var username: String = ""
)
